From 6b3d0fce32f68d9e24e9a958a821c4633c702b6e Mon Sep 17 00:00:00 2001
From: velho <jewel960@gmail.com>
Date: Mon, 25 Dec 2023 03:10:42 +0200
Subject: [PATCH] mg: set tls_mbed impl to v1.3 by default.

Set the minimum and maximum versions to TLSv1.3
to force it to be used by default. Crypto lib initialized
for the mbed TLSv1.3 to work.
---
 src/tls_mbed.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/tls_mbed.c b/src/tls_mbed.c
index f6923616..55f119a2 100644
--- a/src/tls_mbed.c
+++ b/src/tls_mbed.c
@@ -108,6 +108,13 @@ void mg_tls_init(struct mg_connection *c, const struct mg_tls_opts *opts) {
   MG_DEBUG(("%lu Setting TLS", c->id));
   mbedtls_ssl_init(&tls->ssl);
   mbedtls_ssl_config_init(&tls->conf);
+
+  // initialize crypto functionalities to defined state
+  psa_crypto_init();
+
+  mbedtls_ssl_conf_max_tls_version(&tls->conf, MBEDTLS_SSL_VERSION_TLS1_3);
+  mbedtls_ssl_conf_min_tls_version(&tls->conf, MBEDTLS_SSL_VERSION_TLS1_3);
+
   mbedtls_x509_crt_init(&tls->ca);
   mbedtls_x509_crt_init(&tls->cert);
   mbedtls_pk_init(&tls->pk);
-- 
2.25.1

